Patch to remove 3rd party software and replace lapack(e) with openblas and boost 1.55
base on the work of Author: J. Sassmannshausen NHS/GSTT
diff -ruN regenie-3.1.1.orig/Makefile regenie-3.1.1/Makefile
--- regenie-3.1.1.orig/Makefile	2022-05-05 10:59:29.000000000 -0700
+++ regenie-3.1.1/Makefile	2022-06-21 13:34:54.464076000 -0700
@@ -44,7 +44,7 @@
 # detect OS architecture and add flags
 UNAME_S      := $(shell uname -s)
 ifeq ($(UNAME_S),Linux)
- INC          = -I${BGEN_PATH}/3rd_party/boost_1_55_0
+ INC          = -I${BOOST}
  CFLAGS      += -fopenmp
  ifeq ($(strip $(STATIC)),1)
   LPATHS      = -static-libgcc -static-libstdc++
@@ -103,10 +103,10 @@
   INC        += -I${OPENBLAS_ROOT}/include/
   # static linking
   ifeq ($(strip $(STATIC)),1)
-   SLIBS     += -Wl,-rpath=${OPENBLAS_ROOT}/lib/ -llapack -llapacke -lopenblas
+   SLIBS     += -Wl,-rpath=${OPENBLAS_ROOT}/lib/ -lopenblas
   # dynamic linking
   else
-   DLIBS     += -Wl,-rpath=${OPENBLAS_ROOT}/lib/ -llapack -llapacke -lopenblas
+   DLIBS     += -Wl,-rpath=${OPENBLAS_ROOT}/lib/ -lopenblas -lgfortran
   endif
  endif
 endif
@@ -119,11 +119,10 @@
 OBJECTS       = $(patsubst %.cpp,%.o,$(wildcard ./src/*.cpp))
 
 PGEN_PATH     = ./external_libs/pgenlib/
-INC          += -I${PGEN_PATH} -I${PGEN_PATH}/simde/ -I${PGEN_PATH}/include/ -I./external_libs/cxxopts/include/ -I./external_libs/LBFGSpp/include/ -I${BGEN_PATH} -I./external_libs/eigen-3.4.0/ -I${BGEN_PATH}/genfile/include/ -I${BGEN_PATH}/3rd_party/boost_1_55_0/ -I${BGEN_PATH}/3rd_party/zstd-1.1.0/lib -I${BGEN_PATH}/db/include/ -I${BGEN_PATH}/3rd_party/sqlite3 -I./external_libs/
+INC          += -I${PGEN_PATH} -I${PGEN_PATH}/simde/ -I${PGEN_PATH}/include/ -I./external_libs/cxxopts/include/ -I./external_libs/LBFGSpp/include/ -I./external_libs/eigen-3.4.0/ -I${BGEN_PATH}/include/ -I${BOOST}/include -I./external_libs/
 
-LPATHS       += ${LIBMKL} -L${BGEN_PATH}/build/ -L${BGEN_PATH}/build/3rd_party/zstd-1.1.0/ -L${BGEN_PATH}/build/db/ -L${BGEN_PATH}/build/3rd_party/sqlite3/ -L${BGEN_PATH}/build/3rd_party/boost_1_55_0 -L/usr/lib/
 
-LIBS         += ${SLIBS} -lbgen -lzstd -ldb  -lsqlite3 -lboost
+LIBS         += ${SLIBS} -lbgen -lzstd -ldb  -lsqlite3 -lboost_system -lboost_filesystem -lboost_thread -lboost_timer
 LIBS         += -lz ${DLIBS} -lm -ldl -lgfortran
 
 
